R/input-switch.R

```diff
@@ -78,16 +78,48 @@ toggle_switch <- function(id, value = NULL, session = get_current_session()) {
 }
 
 # Keep this internal for now until we agree on the UX
-input_dark_mode_switch <- function(id, label, ..., width = "auto") {
-  res <- input_switch(id, label, ..., width = width)
-  # TODO: .navbar should probably do this for .shiny-input-container
-  res <- tagAppendAttributes(res, style = "margin-bottom: 0;")
-  tagAppendAttributes(
-    res, .cssSelector = "input",
-    onclick = "document.documentElement.setAttribute('data-bs-theme', this.checked ? 'dark' : 'light'); $(window).resize();"
+input_dark_mode_switch <- function(id, ..., mode = NULL) {
+  if (!is.null(mode)) {
+    mode <- rlang::arg_match(mode, c("light", "dark"))
+  }
+
+  if (any(!nzchar(rlang::names2(rlang::list2(...))))) {
+    abort("All arguments in `...` must be named.")
+  }
+
+  res <- web_component(
+    "bslib-dark-mode-switch",
```

Comment:
I think this should be `bslib-input-dark-mode-switch` so that the R function name matches the HTML tag name.

Alternatively, the function could be `dark_mode_switch` and the tag could be `bslib-dark-mode-switch`.

Reply:
We're renaming the R function to `input_dark_mode()` and I'll update this to be consistent, thanks.

Reply:
Done in 4fe55f434da83baa22d91b0652a3d37ef2c28ba9